### PR TITLE
Added Python Solution for Milk Factory

### PR DESCRIPTION
**Pull request by ryanchou-dev**
all test cases pass for this solution

### CODE REVIEW

**Review by nwatx — APPROVED** [review]

**Comment by ryanchou-dev** [comment]
anything else I should add?

File: solutions/usaco-940.mdx [hunk 1]
```diff
@@ -2,11 +2,48 @@
...
+		break
+	# Found a meeting place, remember it.
+	if outgoing[i] == 0: 
+		ans = i
+
+fout = open('factory.out', 'w')
+
+fout.write(str(ans) + '\n')
+
+fout.close()
```
Comment (SansPapyrus683) [inline solutions/usaco-940.mdx:41]:
```suggestion
print(ans, file=open('factory.out', 'w'))
```

File: solutions/usaco-940.mdx [hunk 2]
```diff
@@ -2,11 +2,48 @@
...
+N = int(fin.readline())
+
+outgoing = [0] * 101
+incoming = [0] * 101
+for _ in range(N-1):
+	a, b = map(int, fin.readline().split())
+	# Counting number of outgoing and incoming walkways.
+	outgoing[a] += 1
+	incoming[b] += 1
+ans = -1
```
Comment (SansPapyrus683) [inline solutions/usaco-940.mdx:27]:
```suggestion

ans = -1
```

File: solutions/usaco-940.mdx [hunk 3]
```diff
@@ -2,11 +2,48 @@
...
+## Solution 1
 [Official Analysis](http://www.usaco.org/current/data/sol_factory_bronze_open19.html)
 
+## Implementation
+<LanguageSection>
+<PySection>
+```py
+
+fin = open('factory.in', 'r')
+
```
Comment (SansPapyrus683) [inline solutions/usaco-940.mdx:17]:
```suggestion
```

File: solutions/usaco-940.mdx [hunk 4]
```diff
@@ -2,11 +2,48 @@
...
+N = int(fin.readline())
+
+outgoing = [0] * 101
+incoming = [0] * 101
+for _ in range(N-1):
```
Comment (SansPapyrus683) [inline solutions/usaco-940.mdx:22]: whitespace between operators plz

File: solutions/usaco-940.mdx [hunk 5]
```diff
@@ -2,11 +2,48 @@
...
+	a, b = map(int, fin.readline().split())
+	# Counting number of outgoing and incoming walkways.
+	outgoing[a] += 1
+	incoming[b] += 1
+ans = -1
+for i in range(1, N+1):
+	# Found two meeting places. This won't work.
```
Comment (SansPapyrus683) [inline solutions/usaco-940.mdx:29]: didn't read the statement, but maybe add a phrase or 2 of rationale

File: solutions/usaco-940.mdx [hunk 6]
```diff
@@ -2,11 +2,48 @@
...
+	if outgoing[i] == 0 and ans != -1: 
+		ans = -1
+		break
+	# Found a meeting place, remember it.
```
Comment (SansPapyrus683) [inline solutions/usaco-940.mdx:33]: here as well

File: solutions/usaco-940.mdx [hunk 7]
```diff
@@ -2,11 +2,48 @@
...
+<LanguageSection>
+<PySection>
+```py
+
+fin = open('factory.in', 'r')
+
+N = int(fin.readline())
+
+outgoing = [0] * 101
+incoming = [0] * 101
```
Comment (SansPapyrus683) [inline solutions/usaco-940.mdx:21]: magic numbers are kinda bad
1. are you sure you need statically sized arrays
2. if you 100% need them, put it in a constant called like `MAX_[smth]` idk

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
nitpicking time

File: solutions/usaco-940.mdx (hunk 4 above)
Comment (ryanchou-dev) [reply solutions/usaco-940.mdx:21]: fixin' it. thanks!

File: solutions/usaco-940.mdx (hunk 7 above)
Comment (ryanchou-dev) [reply solutions/usaco-940.mdx]: 👍

File: solutions/usaco-940.mdx [hunk 8]
```diff
@@ -2,11 +2,47 @@
...
 ---
 
+## Solution 1
 [Official Analysis](http://www.usaco.org/current/data/sol_factory_bronze_open19.html)
 
+## Implementation
+<LanguageSection>
+<PySection>
+```py
+
```
Comment (SansPapyrus683) [inline solutions/usaco-940.mdx:15]:
```suggestion
```

**Review by SansPapyrus683 — APPROVED** [review]
good enough for me
just like one more formatting change

**Review by maggie-j-liu — APPROVED** [review]

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
can u add time complexity for both solutions? thanks!

**Comment by ryanchou-dev** [comment]
> can u add time complexity for both solutions? thanks!
👍 (there's one solution btw)

**Comment by HarryWangATX** [comment]
i meant the solution 2, can u just add it as well rq thanks.

**Comment by ryanchou-dev** [comment]
> i meant the solution 2, can u just add it as well rq thanks.

oh, adding it now.

**Comment by ryanchou-dev** [comment]
anything else I should add?

**Review by HarryWangATX — APPROVED** [review]

**Pull request merged**